### PR TITLE
feat(CoachmarkStack): convert to .tsx

### LINKED ISSUE #4232: Add TypeScript types to `@carbon/ibm-products` components
This issue will serve to document and track our work as we provide types to the components we export. We'll start off by tracking this effort into categories of effort required to add types to a particular component. Each component will receive it's own issue so we can chunk this work into bite size pieces.

### LINKED ISSUE #4491: Add Typescript types to CoachmarkStack

### PR DESCRIPTION
**Pull request by emyarod**
Closes #4491 
Related #5097 

This PR converts `CoachmarkStack` and `CoachmarkStackHome` to TypeScript

#### What did you change?

* file extension
* TS interface

#### How did you test and verify your work?

storybook build and type checks

### CODE REVIEW

**Review by makafsal — COMMENTED** [review]
Looks good.

File: packages/ibm-products/src/components/CoachmarkStack/CoachmarkStack.tsx [hunk 1]
```diff
@@ -36,6 +38,70 @@ import { CoachmarkTagline } from '../Coachmark/CoachmarkTagline';
...
 
+type Media =
+  | {
+      render?: () => ReactNode;
+      filePaths?: never;
+    }
+  | {
+      render?: never;
+      filePaths?: string[];
+    };
```
[the remark this answers is not in the document]
Comment (emyarod) [reply packages/ibm-products/src/components/CoachmarkStack/CoachmarkStack.tsx]: I believe that works, thanks for the suggestion!

File: packages/ibm-products/src/components/CoachmarkStack/CoachmarkStack.tsx [hunk 2]
```diff
@@ -36,6 +38,68 @@ import { CoachmarkTagline } from '../Coachmark/CoachmarkTagline';
...
+  /**
+   * The object describing an image in one of two shapes.
+   *
+   * If a single media element is required, use `{render}`.
+   *
+   * If a stepped animation is required, use `{filePaths}`.
+   *
+   * @see {@link MEDIA_PROP_TYPE}.
+   */
+  media: Media;
```
Comment (makafsal) [inline packages/ibm-products/src/components/CoachmarkStack/CoachmarkStack.tsx:75]: As per PropType `media` prop is optional.

File: packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx [hunk 3]
```diff
@@ -16,6 +22,74 @@ import { pkg /*, carbon */ } from '../../settings';
...
+  /**
+   * The object describing an image in one of two shapes.
+   *
+   * If a single media element is required, use `{render}`.
+   *
+   * If a stepped animation is required, use `{filePaths}`.
+   *
+   * @see {@link MEDIA_PROP_TYPE}.
+   */
+  media: Media;
```
Comment (makafsal) [inline packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx:63]: Here also `media` is optional.

File: packages/ibm-products/src/components/CoachmarkStack/CoachmarkStack.tsx (hunk 2 above)
Comment (emyarod) [reply packages/ibm-products/src/components/CoachmarkStack/CoachmarkStack.tsx]: should be updated now!

File: packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx (hunk 3 above)
Comment (emyarod) [reply packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx]: should be updated now!

File: packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx [hunk 4]
```diff
@@ -62,6 +139,24 @@ export let CoachmarkStackHome = forwardRef(
       );
     }
 
+    function renderNavLink(index, label, ref) {
```
Comment (makafsal) [inline packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx:142]: In the previous code ref is optional `ref = null`, can you change it like that?

Comment (emyarod) [reply packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx]: it was already set to `ref ?? null` a few lines below but I've reverted the change

**Review by matthewgallo — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**